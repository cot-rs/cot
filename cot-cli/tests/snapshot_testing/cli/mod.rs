use tempfile::TempDir;

use super::*;
#[test]
fn no_args() {
    assert_cmd_snapshot!(cot_cli!("cli"));
}

#[test]
fn manpages() {
    let tempdir = TempDir::new().unwrap();
    assert_cmd_snapshot!(cot_cli!("cli", "manpages", "-o", tempdir.path()));
}

#[test]
fn completions_missing_shell() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions"));
}

#[test]
fn completions_bash() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "bash"));
}

#[test]
fn completions_elvish() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "elvish"));
}

#[test]
fn completions_fish() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "fish"));
}

#[test]
fn completions_powershell() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "powershell"));
}

#[test]
fn completions_zsh() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "zsh"));
}
