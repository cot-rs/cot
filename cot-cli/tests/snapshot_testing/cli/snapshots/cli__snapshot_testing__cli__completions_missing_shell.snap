---
source: cot-cli/tests/snapshot_testing/cli/mod.rs
info:
  program: cot
  args:
    - cli
    - completions
---
success: false
exit_code: 2
----- stdout -----

----- stderr -----
error: the following required arguments were not provided:
  <SHELL>

Usage: cot cli completions <SHELL>

For more information, try '--help'.
